### PR TITLE
For #18375: Add experiment for set default browser New Tab card.

### PR DESCRIPTION
**Pull request by mcarare**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlController.kt [hunk 1]
```diff
@@ -168,6 +168,18 @@ interface SessionControlController {
...
+     * @see [ExperimentCardInteractor.onSetDefaultBrowserClicked]
+     */
+    fun handleSetDefaultBrowser() {
+    }
+
+    /**
+     * @see [ExperimentCardInteractor.onCloseExperimentCardClicked]
+     */
+    fun handleCloseExperimentCard() {
+    }
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlController.kt:182]: nit: Is there a particular case why we want to have a default implementation?

File: app/src/main/java/org/mozilla/fenix/utils/Settings.kt [hunk 2]
```diff
@@ -292,6 +296,31 @@ class Settings(private val appContext: Context) : PreferencesHolder {
...
+     */
+    fun shouldShowSetAsDefaultBrowserCard(): Boolean {
+        val browsers = BrowsersCache.all(appContext)
+        val experiments = appContext.components.analytics.experiments
+        val isExperimentBranch =
+            experiments.withExperiment(Experiments.DEFAULT_BROWSER) { experimentBranch ->
+                (experimentBranch == ExperimentBranch.DEFAULT_BROWSER_NEW_TAB_BANNER)
+            }
+        return isExperimentBranch &&
+                !userDismissedExperimentCard &&
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/utils/Settings.kt:319]: I wonder if the experiment should only be shown once? As after the experiment has been dismissed one time, we are not going to allow it anymore.

**Review by Amejia481 — COMMENTED** [review]
It looks great, awesome work!

No sure if it's related but the  `org.mozilla.fenix.ui.HistoryTest.deleteAllHistoryTest` UI test is failing.


https://console.firebase.google.com/u/1/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/5171125956912378492/executions/bs.6deed8cbf9b84996/test-cases

File: app/src/main/java/org/mozilla/fenix/utils/Settings.kt (hunk 2 above)
Comment (mcarare) [reply app/src/main/java/org/mozilla/fenix/utils/Settings.kt:319]: According to https://github.com/mozilla-mobile/fenix/issues/18375#issuecomment-815254448 the card should be shown every time after the 3rd app launch until the user dismisses the card.

File: app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlController.kt (hunk 1 above)
Comment (mcarare) [reply app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlController.kt:182]: This is just to follow the pattern used in the app architecture.

Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlController.kt:182]: I asked we the other functions in this interface don't have default implementations.

File: app/src/main/java/org/mozilla/fenix/utils/Settings.kt (hunk 2 above)
Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/utils/Settings.kt:319]: Thanks for the clarification 👍🏽

**Review by Amejia481 — APPROVED** [review]
Awesome work 👍🏽

File: app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlController.kt (hunk 1 above)
Comment (mcarare) [reply app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlController.kt:182]: Oh, sorry, I get it now. I will remove it. :)

**Pull request merged**